rmgpy/data/kinetics/libraryTest.py

```diff
@@ -76,24 +76,25 @@ def test_save_library(self):
         This tests the the library.save method by writing a new temporary file and
         loading it and comparing the original and copied reactions
         """
-        os.makedirs(os.path.join(settings['test_data.directory'],
-                                 'testing_database', 'kinetics', 'libraries', 'eth-oxcopy'))
-        try:
-            self.libraries['ethane-oxidation'].save(
-                os.path.join(settings['test_data.directory'],
-                             'testing_database', 'kinetics', 'libraries', 'eth-oxcopy', 'reactions.py'))
-            self.database.load_libraries(
-                os.path.join(settings['test_data.directory'],
-                             'testing_database', 'kinetics', 'libraries'),
-                libraries=None)  # this loads all of them: ['GRI-Mech3.0', 'ethane-oxidation', 'eth-oxcopy'])
-            ori_rxns = self.database.libraries['ethane-oxidation'].get_library_reactions()
-            copy_rxns = self.database.libraries['eth-oxcopy'].get_library_reactions()
-            for i in range(len(ori_rxns)):
-                if repr(ori_rxns[i]).strip() != repr(copy_rxns[i]).strip():
-                    self.assertIsInstance(copy_rxns[i], TemplateReaction)
-        finally:
-            shutil.rmtree(os.path.join(settings['test_data.directory'],
-                                       'testing_database', 'kinetics', 'libraries', 'eth-oxcopy'))
+        for library_name in ['ethane-oxidation', 'surface-example']:
+            copy_path = os.path.join(settings['test_data.directory'], 'testing_database',
+                                    'kinetics', 'libraries', library_name+'-copy')
+            os.makedirs(copy_path)
+            try:
+                self.libraries[library_name].save(os.path.join(copy_path, 'reactions.py'))
+                self.database.load_libraries(
+                    os.path.join(settings['test_data.directory'], 'testing_database',
+                                'kinetics', 'libraries'),
+                    libraries=None)  # this loads all of them, including the new copy
+                original_rxns = self.database.libraries[library_name].get_library_reactions()
+                copy_rxns = self.database.libraries[library_name+'-copy'].get_library_reactions()
+                for i in range(len(original_rxns)):
+                    if repr(original_rxns[i]).strip() != repr(copy_rxns[i]).strip(): # if it's representation looks different...
+                        self.assertIsInstance(copy_rxns[i], TemplateReaction) # ...it must be a TemplateReaction
+                    #Todo: we need to check more carefully for equivalence, 
```

Comment:
As noted, this test doesn't yet detect that the `metal` attributes go missing when the file is saved.

Reply:
ok so I think what happens is when the library reactions are loaded in, the metal attributes aren't saved at all because the surface info are entry attributes, not reaction attributes. Looking at the debugger, once a library is loaded in, the corresponding entry is None. Not sure if this is a bug when loading in the library itself, but I didn't want metal attributes to belong to a single library because I can see instances where we would want a library with reactions from multiple different metals. This may also be the case where there aren't many database writing tests and we didn't want to be the ones to start. For now, the only time a reaction metal attribute matters is when calculating the reverse rate. After that, I don't believe it is used at all.